coa/pkg/apis/v1alpha2/providers/pubsub/redis/redis.go

```diff
@@ -379,12 +380,35 @@ func toRedisPubSubProviderConfig(config providers.IProviderConfig) (RedisPubSubP
 	if err != nil {
 		return ret, err
 	}
-	err = json.Unmarshal(data, &ret)
-	//ret.Name = providers.LoadEnv(ret.Name)
-	//ret.Host = providers.LoadEnv(ret.Host)
-	//ret.Password = providers.LoadEnv(ret.Password)
-	if ret.NumberOfWorkers <= 0 {
-		ret.NumberOfWorkers = 1
+	var configs map[string]interface{}
+	err = json.Unmarshal(data, &configs)
+	if err != nil {
+		mLog.Info("  P (Redis PubSub): failed to parse to map[string]interface{} %+v", err)
+		return ret, err
+	}
+	configStrings := map[string]string{}
+	for k, v := range configs {
+		configStrings[k] = utils.FormatAsString(v)
+	}
+
+	ret, err = RedisPubSubProviderConfigFromMap(configStrings)
+	if err != nil {
+		mLog.Info("  P (Redis PubSub): failed to parse to RedisPubSubProviderConfig %+v", err)
+		return ret, err
 	}
 	return ret, err
 }
+
+func parseDuration(val string) (time.Duration, error) {
```

Comment:
What is the difference between the parseDuration function and utils.UnmarshalDuration? Seems both try to convert string (no matter it is time str or time float value) to time.Duration? If utils function cannot satisfy our needs, can we fix it?

Reply:
unmarshalDuration is used to unmarshal a marshaled string that contains an integer or string. For example,
"10" -> 10 Nanosecond
"`"10s`"" -> 10s

In our json file, we should support customer to enter both string format "10s" and marshaled strings like "`"10s`"". 
So I change the logic to first call UnmarshalDuration then call parseDuration

Reply:
I have split the change for Redis config to another PR.